web/packages/teleterm/src/mainProcess/contextMenus/tabContextMenu.ts

```diff
@@ -40,38 +46,84 @@ export function subscribeToTabContextMenuEvent(): void {
           return [
             {
               label: 'Close',
-              click: () => resolve(TabContextMenuEventType.Close),
+              click: () => resolve({ event: TabContextMenuEventType.Close }),
             },
             {
               label: 'Close Others',
-              click: () => resolve(TabContextMenuEventType.CloseOthers),
+              click: () =>
+                resolve({ event: TabContextMenuEventType.CloseOthers }),
             },
             {
               label: 'Close to the Right',
-              click: () => resolve(TabContextMenuEventType.CloseToRight),
+              click: () =>
+                resolve({ event: TabContextMenuEventType.CloseToRight }),
             },
           ];
         }
 
         function getPtyTemplate(): MenuItemConstructorOptions[] {
           if (
-            options.documentKind === 'doc.terminal_shell' ||
-            options.documentKind === 'doc.terminal_tsh_node'
+            options.document.kind === 'doc.terminal_shell' ||
+            options.document.kind === 'doc.terminal_tsh_node'
+          ) {
+            return [
+              {
+                label: 'Duplicate Tab',
+                click: () =>
+                  resolve({ event: TabContextMenuEventType.DuplicatePty }),
+              },
+            ];
+          }
+        }
+
+        function getShellTemplate(): MenuItemConstructorOptions[] {
+          const doc = options.document;
+          if (
+            doc.kind === 'doc.terminal_shell' ||
+            doc.kind === 'doc.gateway_kube'
           ) {
+            const activeShell = doc.shellId;
+            const defaultShell = configService.get('terminal.shell').value;
             return [
               {
                 type: 'separator',
               },
               {
-                label: 'Duplicate Tab',
-                click: () => resolve(TabContextMenuEventType.DuplicatePty),
+                label: `Active Shell (${activeShell})`,
+                type: 'submenu',
+                submenu: shells.map(shell => ({
+                  label: shell.friendlyName,
+                  id: shell.id,
+                  type: 'radio',
+                  checked: shell.id === activeShell,
+                  click: () => {
+                    resolve({
+                      event: TabContextMenuEventType.ReopenPtyInShell,
+                      item: shell,
+                    });
+                  },
+                })),
+              },
+              {
+                label: `Default Shell (${defaultShell})`,
+                type: 'submenu',
+                submenu: shells.map(shell => ({
+                  label: shell.friendlyName,
+                  id: shell.id,
+                  type: 'radio',
+                  checked: shell.id === defaultShell,
+                  click: () => {
+                    configService.set('terminal.shell', shell.id);
```

Comment:
Changing default shell is a bit wonky, I believe that's because we cache the default shell when the app first launches. There's also a blank line above the message.


https://github.com/user-attachments/assets/ae5fe3f8-13b1-45a8-ae74-b4d66bb680c8

Reply:
What exactly do you mean? That the active shell doesn't change when you change the default shell?

Reply:
In the video, I change the active shell from zsh to bash. But the message in the terminal says "The default interactive shell is now zsh". That "is now" part makes it seems like I just changed the default shell, but I did not do that.

Then I change the default shell to bash and open a new terminal tab. I once again see the same message, even though I've just changed the default shell from zsh to bash.

Reply:
Ah, I see, this message is unfortunate here. But the fix should be easy, we need to set the env var `BASH_SILENCE_DEPRECATION_WARNING=1`.